docs/reference/ccr/overview.asciidoc

```diff
@@ -110,3 +110,50 @@ your use case.
 You can terminate replication with the
 {ref}/ccr-post-unfollow.html[unfollow API]. This API converts a follower index
 to a regular (non-follower) index.
+
+[float]
+=== Upgrading clusters
+
+In case of upgrading clusters, clusters that are actively using CCR, require a slightly different upgrade approach.
+Otherwise index following may fail during a rolling upgrade, because of the following reasons:
+
+* If a new index setting or mapping type is replicated from an upgraded cluster
+  to a not upgraded cluster then the upgraded cluster will reject that and will
+  fail index following.
+* Lucene is not forwards compatible and when index following is falling back to
+  file based recovery then a node in a not upgraded cluster will reject index files
+  from a newer Lucene version compared to what it is using.
+
+Rolling upgrading clusters with CCR is different in case of unidirectional index following and
+bidirectional index following.
+
+[float]
+==== Unidirectional index following
+
+In this kind of a setup a cluster contains either only leader indices or
```

Comment:
In a uni-directional setup between two clusters, one cluster contains only leader indices, and the other cluster contains only follower indices following indices in the first cluster.

Reply:
Let us strike this sentence:

>For example one cluster contains all the leader indices and one or more clusters only contain the follower indices.

so that we can keep the example focused on two clusters, as it's simpler to understand. The final paragraph about chain following is good, as it helps then generalize the simple case.